consensus/participant.go

```diff
@@ -65,9 +70,20 @@ func NewParticipant(
 		return nil, fmt.Errorf("could not initialize timeout config: %w", err)
 	}
 
+	// register as dynamically updatable via admin command
+	if cfg.Registrar != nil {
+		err = cfg.Registrar.RegisterDurationConfig("hotstuff-block-rate-delay", timeoutConfig.GetBlockRateDelay, timeoutConfig.SetBlockRateDelay)
```

Comment:
Hm, was this tested? I am not sure how it will work with `timeoutConfig` as value, here we will register callback but later when we pass it to `NewController` we will make another copy, won't we? This will be another object and in that case it shouldn't be reflected. Unless I am missing something.

Reply:
Seems like this has been in place since we deployed in January. I have definitely used this admin command on Mainnet and it works. Looking at it, it should work because the field being updated is a pointer within `Config`, so that reference is maintained even when `Config` is copied. 

I agree it's non-obvious why it works though. We could switch to `*Config` or add a test case documenting the behaviour.

Reply:
Thanks @jordanschalm, right, if it updates pointer internally then it has to work, I would definitely prefer if `Config` would be pointer because as you said, it's not obvious.

Reply:
Unfortunately, changing to a pointer generates quite a lot of changes. In particular, we have other configs, that embed `timeout.Config` that also would need to be changed. 

For the scope of this PR, I opted to
* add documentation 
* add unit tests 

👉  https://github.com/onflow/flow-go/pull/4234/commits/0c09c65b983f660c3d2b3177b5daffe83da669d5